modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/SearchAsYouTypeFieldMapper.java

```diff
@@ -376,7 +376,7 @@ public Query prefixQuery(String value, MultiTermQuery.RewriteMethod method, bool
 
         @Override
         public ValueFetcher valueFetcher(MapperService mapperService, SearchLookup searchLookup, String format) {
-            throw new UnsupportedOperationException();
+            return SourceValueFetcher.toString(name(), mapperService, format);
```

Comment:
I think these should be fetching the value from their parent field?  `name()` is going to return the subfield name here, which won't be in the source.